packages/atlas-service/src/secret-store.ts

```diff
@@ -1,28 +1,40 @@
-import keytar from 'keytar';
-import { app } from 'electron';
+import { UserData, z } from '@mongodb-js/compass-user-data';
+import { safeStorage } from 'electron';
 
-export const SECRET_STORE_KEY = 'AtlasLoginOIDCPluginState';
+const AtlasPluginStateSchema = z.string().optional();
 
 export class SecretStore {
-  async getItem(key: string): Promise<string | undefined> {
+  private readonly userData: UserData<typeof AtlasPluginStateSchema>;
+  private readonly fileName = 'AtlasPluginState';
+  constructor(basePath?: string) {
+    this.userData = new UserData(AtlasPluginStateSchema, {
+      subdir: '',
+      basePath,
+    });
+  }
```

Comment:
Now that we are saving this data on fs, already logged in users will have to login again as we are not migrating the data (from keytar). I think its fine to keep it that way instead of writing a migration (and eventually coming back to remove the code).

If you feel strongly otherwise, let me know.

Reply:
I'll check with product.